core/drivers/imx_i2c.c

```diff
@@ -0,0 +1,380 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * (c) 2020 Jorge Ramirez <jorge@foundries.io>, Foundries Ltd.
+ */
+#include <arm.h>
+#include <io.h>
+#include <drivers/imx_i2c.h>
+#include <kernel/delay.h>
+#include <kernel/generic_boot.h>
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
+#include <registers/imx8m-crm_regs.h>
+#include <registers/imx8m.h>
+#include <stdlib.h>
+#include <trace.h>
+#include <util.h>
+
+static struct io_pa_va i2c_bus[] = {
+	{ I2C1_BASE_ADDR, 0, },
```

Comment:
better if specifies the structure fields.
no need to defined `.va=0`.
```c
static struct io_pa_va i2c_bus[] = {
	{ .pa = I2C1_BASE_ADDR },
	{ .pa = I2C2_BASE_ADDR },
	{ .pa = I2C3_BASE_ADDR },
```

Reply:
> A comment that I leave up to iMX maintainers to comment on: this looks like an i2c driver specific to imx8m hence I think the source file should be renamed t **imx8m_i2c.c**. The header looks fine for a generic **imx_i2c.h** naming.

no. this is not an imx8m specific driver. it just expects other users to research and add the necessary addresses to the relevant header file.